src/cyclefold/gadgets.rs

```diff
@@ -0,0 +1,586 @@
+use super::nova_circuit::FoldingData;
+
+use crate::{
+  gadgets::{
+    ecc::AllocatedPoint,
+    r1cs::{AllocatedR1CSInstance, AllocatedRelaxedR1CSInstance},
+  },
+  traits::{commitment::CommitmentTrait, Engine, ROCircuitTrait},
+};
+
+use bellpepper_core::{ConstraintSystem, SynthesisError};
+pub struct AllocatedFoldingData<E: Engine> {
+  pub U: AllocatedRelaxedR1CSInstance<E>,
+  pub u: AllocatedR1CSInstance<E>,
+  pub T: AllocatedPoint<E>,
+}
```

Comment:
This `FoldingData` is made up of the "correct" field instances and points. It could even be used in the original Nova/SuperNova augmented circuit if we wanted. I added it here because it reduces a lot of code duplication from the data in the augmented circuit coming from the multiple invocations of the Cyclefold circuit.